src/gui/render.cpp

```diff
@@ -140,9 +144,9 @@ static void empty_line_handler(const void*) {}
 static void start_line_handler(const void* s)
 {
 	if (s) {
-		auto src   = static_cast<const uintptr_t*>(s);
+		auto src = static_cast<const uintptr_t*>(s);
 		auto cache = reinterpret_cast<uintptr_t*>(render.scale.cacheRead);
-		for (Bits x = render.src.start; x > 0;) {
+		for (Bits x = render.src_start; x > 0;) {
```

Comment:
maybe similar to your capture code, and we can replace the `Bits x` with:

```c++
auto pixels_to_render = render.src_start;
while (pixels_to_render--) {
  ...
}
```

Reply:
Existing code, don't wanna refactor it now. I just had to touch it because I had to after introducing the RenderParams struct.

Reply:
Understandable :+1: